File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d1,d2={},{}
        for i in s:
            if i not in d1:
                d1[i] = 1
            else:
                d1[i] += 1
        for i in t:
            if i not in d2:
                d2[i] = 1
            else:
                d2[i] += 1
        if(len(s)>=len(t)):
            for i in d1:
                if i not in d2 or d1[i]!=d2[i]:
                    return False
            return True
        else:
            for i in d2:
                if i not in d1 or d1[i]!=d2[i]:
                    return False
            return True

File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int> mp1, mp2;
        for(auto i:s){
            if(mp1.find(i)!=mp1.end())
                mp1[i] += 1;
            else
            mp1[i] = 1;
        }
        for(auto i:t){
            if(mp2.find(i)!=mp2.end())
                mp2[i] += 1;
            else
            mp2[i] = 1;
        }
        bool val = true;
        if(mp2.size()>=mp1.size())
        for(auto i: mp2){
            if(mp1.find(i.first)==mp1.end() || mp1[i.first]!=mp2[i.first])
                val = false;
        }
        else
            for(auto i: mp1){
            if(mp2.find(i.first)==mp2.end() || mp1[i.first]!=mp2[i.first])
                val = false;
        }
        return val;
    }
};